### ISSUE TITLE
Random crashes when opening quick actions

### ISSUE THREAD

**Issue by Galarzaa90** [EWSoftware/VSSpellChecker#151]
When  opening quick actions (right clicking on light bulb or Alt+Enter), sometimes Visual Studio crashes entirely. This only happens on lines that have spelling errors.

I love this plugin, but to use it, I have to save every single change I make because I might get a crash at any time.

## Versions info:
Visual Studio Community 2017 15.5.5
.NET Framework 4.7.02556
Visual Studio Spell Checker 2017.12.29.0 (downloaded from [here](https://marketplace.visualstudio.com/items?itemName=EWoodruff.VisualStudioSpellCheckerVS2017andLater))

## System info:
Windows 10 Home Single Language 10.0.16299

Need any logs files? Let me know, I'm not sure where to look.

**Comment by EWSoftware** [comment]
I haven't heard of or experienced this issue before.  If you can supply a log file it might help me see where it's crashing.

**Comment by Galarzaa90** [comment]
This is all I could find:
https://gist.github.com/Galarzaa90/1fb0230176b891839a663f87ecc5168d

Is there another place to look for logs?

I got this running VisualStudio with the `/Log` argument.

**Comment by EWSoftware** [comment]
Unfortunately, that doesn't help as it doesn't list the spell checker extension anywhere in the log as a possible cause.  You can try doing a repair on Visual Studio and updating to the latest release if you haven't done so already to see if that corrects the problem.

If you can reproduce it reliable and can supply an example of how to do so, I can try to recreate the problem.

**Comment by sharwell** [comment]
:link: Also reported here:
[[crash] Accrocher l'ampoule VS 2017-[crash] Hang the bulb VS 2017](https://developercommunity.visualstudio.com/content/problem/322451/crash-accrocher-lampoule-vs-2017.html)

@EWSoftware I will provide more information to help resolve this using the information provided by the user.

**Comment by sharwell** [comment]
I'm guessing it's caused by building against an unspecified build of the Visual Studio SDK, resulting in an undeclared dependency on some patch version of Visual Studio 2017 more recent than 15.0.

https://github.com/EWSoftware/VSSpellChecker/blob/986f35c3281886134a448217b7b3f5f785e4ce30/Source/VSSpellChecker/VSSpellChecker2017AndLater.csproj#L536-L544

The user in this case is running Visual Studio 2017 version 15.2. Likely the following approaches are viable:

* Restrict the VSIX installation target to the correct minimum version, which under the current configuration is the same build installed on the machine used to create the VSIX
* Use NuGet package references for the VSSDK instead of relying on images installed in the current version

**Comment by EWSoftware** [comment]
I've made changes to support async package loading for the next release.  Part of that was recreating the Visual Studio project from scratch so that it uses the NuGet packages rather than the actual references so that I can stay a few versions behind.  This was also done in an attempt to help resolve the reported issues with it not finding the proper version of Microsoft.VisualStudio.Threading (#156 and #174).

Is there a reference somewhere for the version numbers to use in the VSIX to restrict it to a minimum version of VS2017?  It didn't seem to be tied to the versions in the About box (15.6, 15.7, 15.8.2. etc.).  The last time I tried it was with 15.2 as I recall and specifying that wouldn't let me install it on that version on my build PC so that's why I've left it set to 15.0.
